src/MetadataManager.actor.cpp

```diff
@@ -166,23 +162,20 @@ ACTOR static Future<std::pair<Reference<UnboundCollectionContext>, uint64_t>> co
 }
 
 ACTOR static Future<Reference<UnboundCollectionContext>> assembleCollectionContext(Reference<DocTransaction> tr,
-                                                                                   std::string dbName,
-                                                                                   std::string collectionName,
+                                                                                   Namespace ns,
```

Comment:
Consider taking a reference, rather than a copy?

Reply:
Its an `ACTOR`, because of Actor Compiler tricks, `Namespace ns` would become `Namespace const& ns`. If I make it `Namespace & ns` that would confuse Actor compiler. Basically, for any ACTOR function, we are forced to pass by value or use `Reference<>`